### ISSUE TITLE
Difficulty canceling sub-flow runs created by run_deployment() from within a parent flow.

### ISSUE THREAD

**Issue by taylor-curran** [PrefectHQ/prefect#9738]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

Reporting cancellation issues from a user, they reported:

We experience some trouble when trying to cancel sub-flow runs created by run_deployment() from within a parent flow. As a quick background, we use run_deployment() as the payload function for our custom StartFlowRun Tasks to create subflows from deployments of other flows. We use Concurrent task runner + K8s infrastructure blocks, which according to the documentation both supports cancellation. https://docs.prefect.io/latest/concepts/flows/#cancel-a-flow-run

Below is our operation  that leads to the problem:

1. After the subflow run is created, we tried to cancel the parent flow and all its subflows. The parent flow run was cancelled easily with the “Cancel” button, but when we try to cancel the subflows, we realized there is no “Cancel” button for the subflow run.

2. Noting that we can also use CLI for flow run cancellation, we attempted to cancel the sub-flow run via CLI, and the CLI operation succeeded, noting flow run has been cancelled. Based on the documentation of flow run cancellation, we assumed this means the infrastructure will be killed in 30 seconds.

3. Later yesterday we received email notifications of failures in the downstream task runs of the supposedly cancelled sub-flow run. This is particularly self-contradicting:

3-i. The sub-flow run page says that this subflow has been cancelled, with last updated time = the time we tried to cancel everything:

![image](https://github.com/PrefectHQ/prefect/assets/68969861/54321140-0335-4f95-9c6b-a7ccb2e34181)

3-ii. The task runs in this flow ran on for 10 hours, erroring out much later:
![image](https://github.com/PrefectHQ/prefect/assets/68969861/e8cc8fe5-c24b-4503-a7c9-864de56d3f5e)

Based on the observation above, we have the following questions:

1. Why is subflows created with run_deployment missing a Cancel button, especially since they are equivalent to starting a flow run from deployment in the UI anyways? We noticed there had been some GitHub issues related to removing cancellation button for subflow runs, but we think the case above does not fit into the description in: https://github.com/PrefectHQ/prefect/issues/7736

2. When we cancel the parent flow run, we did not see any sub-flow runs being cancelled automatically, but why does the state message in the screenshot above says the reason of cancellation is because “The parent flow run was cancelled”? Is there a default behavior for all sub flow runs to be cancelled with the parent flow run? If so, why is it not working here?

3. Why would the subflow run enters a Cancelled State while its infrastructure is still running? Does this contradicts to the prior that Prefect 2’s cancellation functionality is robust in a sense that the infrastructure will guaranteed to be killed?

### Reproduction

```python3
See steps above.
```

### Versions

```Text
2.9.0
```

**Comment by madkinsz** [comment]
Can you use > instead of ` for these quotes? Otherwise we need to scroll horizontally to read them.

**Comment by PPPSDavid** [comment]
Hi, I would like to provide some more context to the problem and see if there is any development on this:

Recently when we try to cancel a flow run with subflow launched by run_deployment on a K8s infrastructure, we noticed that:

parent flow is cancelled and pod killed( expected)

subflow is set to a cancelled state by the cancellation cleanup, but the pod is still running. This caused weird behavior during retry, since if I retry the subflow, **there will actually be 2 pods running.**

**Comment by PPPSDavid** [comment]
Hi Prefect, bumping this, has there been any update?

**Comment by zhen0** [comment]
For the first part of this issue, https://github.com/PrefectHQ/prefect-ui-library/pull/1453 should enable cancellation of runs from run_deployment() from the UI. 

For the others I want to check the issue is still cancelling from a parent flow run?  Or is it directly cancelling the run_deployment run itself?  We did some updates to cancellation so it would be useful to have this issue narrowed down. 

I'm adding to the backlog as I think there's some neatening that still needs to happen here.

**Comment by PPPSDavid** [comment]
Hi @zhen0 ,Thanks for the update! Yes, the issue occurred earlier this month when **we tried to cancel the parent flow run directly.** 

what we did:

1. A flow of subflows is running
2. We hit cancel button on the parent flow run
3. Both flow runs are marked as cancelled
4. After a while, when we **restart** the subflow run and the parent flow run, we noticed duplicated K8s pods for the **sub flow run only** (the parent flow’s infrastructure has been killed as expected by the cancellation).
